.travis.yml

```
@@ -22,17 +21,23 @@ install:
   - pip install -r requirements.txt
   - pip install codecov
   - pip install pytest-cov
+  - pip install mypy
 
 env:
   - BB_UNIONFS_ENABLE=false BB_PLUGINS_EXPERIMENTS='["benchbuild.experiments.raw", "benchbuild.experiments.empty"]'
 
 script:
-  - coverage run -a `which benchbuild` bootstrap -s
-  - pytest --cov=./ benchbuild tests
-  - coverage run -a `which benchbuild` config view
-  - coverage run -a `which benchbuild` project view
-  - coverage run -a `which benchbuild` experiment view
-  - coverage run -a `which benchbuild` -vvvvv run --full test
+  - coverage run -p `which benchbuild` bootstrap -s
```

```suggestion
  - pre-commit run --from-ref origin/master --to-ref HEAD
  - coverage run -p `which benchbuild` bootstrap -s
```

If this passes for BB :wink:

---

pre-commit is not active on master yet

---

It does not have to be, this just tells pre-commit only to check the diff between the current HEAD and master, i.e., only complaining if the PR has bad patterns but not if the rest of the code base has them.

---

This requires pre-commit to be available in the travis build environment. So either it is installed in .travis.yml or it is provided by requirements.txt

Neither is true.

---

kk than add it later